org-roam-db.el

```diff
@@ -276,7 +276,7 @@ If UPDATE-P is non-nil, first remove the file in the database."
   "Return t if headline at point is a node, else return nil."
   (and (org-id-get)
        (not (cdr (assoc "ROAM_EXCLUDE" (org-entry-properties))))
-       (funcall org-roam-db-node-exclude-function)))
+       (funcall org-roam-db-node-include-function)))
```

Comment:
since this is inside an `and` with 2 other predicates, `org-roam-db-node-include-function` is not enough to guarantee the node is included. A clearer expression of the behaviour would be: 

```
(not (funcall org-roam-db-node-exclude-function))))
```

Reply:
@jethrokuan  using the `org-roam-db-node-include-function` to exclude org-attach headings, I get a failure rebuilding the cache. There is a FK constraint violation in `links.source` → `nodes.id`. This occurs when it tries to insert the org-attach link from the heading to the attached file because the heading must be present in the `nodes` table.

Reply:
Here I wanted `ROAM_EXCLUDE` to always work, and also all nodes need an ID. so they take precedence over the include function.

Reply:
That precedence would still hold with the logic I'm proposing:

```
  (and (org-id-get)
       (not (cdr (assoc "ROAM_EXCLUDE" (org-entry-properties))))
       (not (funcall org-roam-db-node-exclude-function)))
```

Which comes down to `(and has-id  (not is-excluded))` where `is-excluded` is `(or has-ROAM_EXCLUDE is-org-roam-db-node-excluded)`. It just feels closer to the intended semantics in the config file, and draws up a nice parallel with `ROAM_EXCLUDE`, making the final behavior easier to predict by the user:

```
(setq org-roam-db-node-exclude-function
      (lambda ()
        (member "ATTACH" (org-get-tags))))
```